backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/expiry/AstractMonthsUpdateStrategy.java

```diff
@@ -0,0 +1,109 @@
+package de.metas.handlingunits.expiry;
+
+import de.metas.handlingunits.HuId;
+import de.metas.handlingunits.IHUContext;
+import de.metas.handlingunits.IHandlingUnitsBL;
+import de.metas.handlingunits.IHandlingUnitsDAO;
+import de.metas.handlingunits.IMutableHUContext;
+import de.metas.handlingunits.attribute.storage.IAttributeStorage;
+import de.metas.handlingunits.model.I_M_HU;
+import de.metas.organization.ClientAndOrgId;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.compiere.util.Env;
+
+import java.util.Iterator;
+
+/*
+ * #%L
+ * de.metas.handlingunits.base
+ * %%
+ * Copyright (C) 2019 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+public abstract class AstractMonthsUpdateStrategy
+{
+	// services
+	final IHandlingUnitsBL handlingUnitsBL;
+	final IHandlingUnitsDAO handlingUnitsDAO = Services.get(IHandlingUnitsDAO.class);
+
+	protected AstractMonthsUpdateStrategy(@NonNull final IHandlingUnitsBL handlingUnitsBL)
+	{
+		this.handlingUnitsBL = handlingUnitsBL;
+	}
+
+	public abstract Iterator<HuId> getAllSuitableHUs();
```

Comment:
instead of having that abstract method there, why not having it as a ctor/method param ?

Reply:
reading ur code again, i have the feeling u don't need AstractMonthsUpdateStrategy but u need those 2 actual implementations, and somewhere refactored some static helper methods.